compiler/src/dotty/tools/backend/sjs/JSCodeGen.scala

```diff
@@ -785,19 +785,37 @@ class JSCodeGen()(using genCtx: Context) {
       !f.isOneOf(Method | Module) && f.isTerm
         && !f.hasAnnotation(jsdefn.JSNativeAnnot)
         && !f.hasAnnotation(jsdefn.JSOptionalAnnot)
-    }.map({ f =>
+    }.flatMap({ f =>
       implicit val pos = f.span
 
-      val flags = js.MemberFlags.empty.withMutable(f.is(Mutable))
+      val isStaticField = f.is(JavaStatic).ensuring(isStatic => !(isStatic && isJSClass))
+
+      val namespace = if isStaticField then js.MemberNamespace.PublicStatic else js.MemberNamespace.Public
+      val mutable = isStaticField || f.is(Mutable)
+
+      val flags = js.MemberFlags.empty.withMutable(mutable).withNamespace(namespace)
 
       val irTpe =
         if (isJSClass) genExposedFieldIRType(f)
         else toIRType(f.info)
 
       if (isJSClass && f.isJSExposed)
-        js.JSFieldDef(flags, genExpr(f.jsName)(f.sourcePos), irTpe)
+        js.JSFieldDef(flags, genExpr(f.jsName)(f.sourcePos), irTpe) :: Nil
       else
-        js.FieldDef(flags, encodeFieldSym(f), originalNameOfField(f), irTpe)
+        val fieldDef = js.FieldDef(flags, encodeFieldSym(f), originalNameOfField(f), irTpe)
+        val rest =
+          if isStaticField then
```

Comment:
We should add a comment somewhere around here explaining that this generates static getters for static fields, and that this is necessary notably for enums.

Reply:
I've added a comment